### PR TITLE
Address prototype comments in file types branch

### PR DESCRIPTION
**Pull request by RikkiGibson**
Related to #60819

### CODE REVIEW

File: src/Compilers/CSharp/Portable/Errors/ErrorCode.cs [hunk 1]
```diff
@@ -2099,12 +2099,12 @@ internal enum ErrorCode
         ERR_FixedFieldMustNotBeRef = 9049,
         ERR_RefFieldCannotReferToRefStruct = 9050,
 
-        // PROTOTYPE(ft): compress these before feature merge
-        ERR_FileTypeDisallowedInSignature = 9300,
-        ERR_FileTypeNoExplicitAccessibility = 9301,
-        ERR_FileTypeBase = 9302,
-        ERR_FileTypeNested = 9303,
-        ERR_GlobalUsingStaticFileType = 9304
+        ERR_FileTypeDisallowedInSignature = 9351,
```
Comment (cston) [inline src/Compilers/CSharp/Portable/Errors/ErrorCode.cs:2102]: >9351 [](http://example.com/codeflow?start=44&length=4)

9051, ...?

File: src/Compilers/CSharp/Test/Syntax/Parsing/FileModifierParsingTests.cs [hunk 2]
```diff
@@ -2377,7 +2428,6 @@ void M()
         EOF();
     }
 
-    // PROTOTYPE(ft): confirm whether this breaking change is acceptable
```
Comment (cston) [inline src/Compilers/CSharp/Test/Syntax/Parsing/FileModifierParsingTests.cs:2380]: >PROTOTYPE(ft): confirm whether this breaking change is acceptable [](http://example.com/codeflow?start=7&length=65)

Was this a stale comment? It looks like we report these binding errors today.

**Comment by cston** [comment]
>     public void TopLevelVariable_01()

Is this behavior identical to `-langversion:10`? If so, consider merging into one test with `[Theory]` so it's clear.

---
Refers to: src/Compilers/CSharp/Test/Syntax/Parsing/FileModifierParsingTests.cs:2474 in 7eaa444. [](commit_id = 7eaa4440d01acb3025885516e2c3445afc5d3132, deletion_comment = False)

File: src/Compilers/CSharp/Test/Syntax/Parsing/FileModifierParsingTests.cs [hunk 3]
```diff
@@ -2500,33 +2551,17 @@ public void TopLevelVariable_02()
     [Fact]
```
Comment (cston) [inline src/Compilers/CSharp/Test/Syntax/Parsing/FileModifierParsingTests.cs:2551]: >Fact [](http://example.com/codeflow?start=5&length=4)

Consider testing with:
```
[Theory]
[InlineData(LanguageVersion.CSharp10)]
[InlineData(LanguageVersionFacts.CSharpNext)]
```

File: src/Compilers/CSharp/Test/Syntax/Parsing/FileModifierParsingTests.cs (hunk 2 above)
Comment (RikkiGibson) [reply src/Compilers/CSharp/Test/Syntax/Parsing/FileModifierParsingTests.cs:2380]: I think whether 'file' in a type position parses as a type depends on context, and for simplicity we disallow you from using it as a type name. It ends up continuing to parse as a type in this scenario. I'll update the breaking change doc.

File: src/Compilers/CSharp/Portable/Errors/ErrorCode.cs (hunk 1 above)
Comment (RikkiGibson) [reply src/Compilers/CSharp/Portable/Errors/ErrorCode.cs]: Fixed

File: src/Compilers/CSharp/Test/Syntax/Parsing/FileModifierParsingTests.cs (hunk 3 above)
Comment (RikkiGibson) [reply src/Compilers/CSharp/Test/Syntax/Parsing/FileModifierParsingTests.cs]: Fixed this and the similar previous comment.

**Comment by cston** [comment]
>     public void Variable_01()

Was this test redundant?

---
Refers to: src/Compilers/CSharp/Test/Syntax/Parsing/FileModifierParsingTests.cs:2610 in 7eaa444. [](commit_id = 7eaa4440d01acb3025885516e2c3445afc5d3132, deletion_comment = True)

**Review by cston — APPROVED** [review]

**Comment by RikkiGibson** [comment]
> > ```
> > public void Variable_01()
> > ```
> 
> Was this test redundant?
> 
> Refers to: src/Compilers/CSharp/Test/Syntax/Parsing/FileModifierParsingTests.cs:2610 in [7eaa444](https://github.com/dotnet/roslyn/commit/7eaa4440d01acb3025885516e2c3445afc5d3132). [](commit_id = [7eaa444](https://github.com/dotnet/roslyn/commit/7eaa4440d01acb3025885516e2c3445afc5d3132), deletion_comment = True)

I think it was not redundant. I got mixed up while I was going through the feedback. Have restored the test.

File: docs/compilers/CSharp/Compiler Breaking Changes - DotNet 7.md [hunk 4]
```diff
@@ -427,3 +427,16 @@ class @required {} // No error
 This was done as `required` is now a member modifier for properties and fields.
 
 You can learn more about this change in the associated [csharplang issue](https://github.com/dotnet/csharplang/issues/3630).
+
+## Types cannot be named `file`
```
Comment (jcouv) [inline docs/compilers/CSharp/Compiler Breaking Changes - DotNet 7.md:431]: Please place the new entry at the top. We changed the order when mirroring process to docs.microsoft.com started (was done in [https://github.com/dotnet/roslyn/pull/60260](https://github.com/dotnet/roslyn/pull/60260))

File: src/Compilers/CSharp/Test/Syntax/Parsing/FileModifierParsingTests.cs [hunk 5]
```diff
@@ -1434,6 +1463,54 @@ record file { }
...
 
+    [Fact]
+    public void TypeNamedFile_01()
+    {
+        UsingNode($$"""
+            class file { }
+            """,
+            expectedBindingDiagnostics: new[]
+            {
+                // (1,7): error CS9056: Types and aliases cannot be named 'file'.
```
Comment (jcouv) [inline src/Compilers/CSharp/Test/Syntax/Parsing/FileModifierParsingTests.cs:1474]: Let's also test this scenario (type named "file") with C# 10

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 8) modulo fixing the order in breaking changes

**Comment by RikkiGibson** [comment]
Verified that the most recent changes pass locally and will verify in CI when merging to main.

**Pull request merged**